stock_mts_mto_mrp/model/mrp_production.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018 Vauxoo, www.vauxoo.com
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from odoo.exceptions import UserError
+
+
+class MrpProduction(models.Model):
+    """ Manufacturing Orders """
+    _inherit = 'mrp.production'
+
+    @api.multi
+    def _adjust_procure_method(self):
+        """This method is taken from odoo as is from the module mrp. This is
+        done because there was no way to interfiere in the decision making
+        about what routes to take"""
+        try:
+            mto_route = self.env['stock.warehouse']._get_mto_route()
+        except UserError:
+            mto_route = self.env['stock.location.route']
+        # This new route is for taking in count this module, the rest is the same as odoo # noqa
+        mto_route |= self.env['stock.location.route'].search([
```

Comment:
another important point, use the translation like the original method:
https://github.com/odoo/odoo/blob/717b7249d5d47421894052e8ad8dc67bc6f3603f/addons/stock/models/stock_warehouse.py#L479

Reply:
I will do(try to make) the fix so adding another another route similar to mto will work, I think that your reviews will make an point in that PR... but I will not make an enhance to their efficiency...

Reply:
I will do(try to make) the fix so adding another another route similar to mto will work, I think that your reviews will make an point in that PR... but I will not make an enhance to their efficiency(at the moment)...

Reply:
Just an update, didn't make that PR because we have deeper problems with the logic of MTS+MTO, odoo is not fit(yet) to store this method, so, after adding this fix to one of our methods we found that we got another problem:

The materials with MTS+MTO in a product aren't going to ever touch our module, so, the decision between stock and purchase is never triggered. If we added the logic that this action is triggered in MRP we enter into a loop...right now I'm out of ideas and the team is looking forward a solution.

Reply:
Hi, What's the difference between this PR and this : https://github.com/OCA/manufacture/tree/11.0/mrp_mto_with_stock/
In my mind, with mrp_mto_with_stock the job is done, but you need to clic on "check avaibility" on a MO, and it's very hard when you have a lot of MO.
Why don't you improve mrp_mto_with_stock ?